spec/requests/headers_spec.rb

```diff
@@ -24,4 +24,10 @@
 
     expect(response.code.to_i).to eq(404)
   end
+
+  it 'handles headers that are not ASCII encodable' do
+    get root_path, headers: { 'Host' => '¿/?' }
```

Comment:
this might be the wrong header to test because we delete the entire header?

Reply:
I changed up this spec so that it doesn't use the Host header. Mind taking another look?